kie-soup-dataset/kie-soup-dataset-sql/src/test/java/org/dashbuilder/dataprovider/sql/SQLDataSetProviderTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dashbuilder.dataprovider.sql;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.atLeastOnce;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.dashbuilder.dataprovider.sql.dialect.Dialect;
+import org.dashbuilder.dataprovider.sql.model.Column;
+import org.dashbuilder.dataset.ColumnType;
+import org.dashbuilder.dataset.DataSetMetadata;
+import org.dashbuilder.dataset.def.DataSetDefFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.modules.junit4.PowerMockRunnerDelegate;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(Parameterized.class)
+@PrepareForTest({JDBCUtils.class})
+public class SQLDataSetProviderTest {
+
+    @Parameters(name = "Dialect : {0}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                 {JDBCUtils.DB2, "FETCH FIRST 0 ROWS ONLY"}, 
+                 {JDBCUtils.DEFAULT, "LIMIT 0"},
+                 {JDBCUtils.H2, "LIMIT 0"},
+                 {JDBCUtils.MYSQL, "LIMIT 0"},
+                 {JDBCUtils.ORACLE, "FETCH FIRST 0 ROWS ONLY"},
+                 {JDBCUtils.POSTGRES, "LIMIT 0"},
+                 {JDBCUtils.SQLSERVER, "TOP 0"},
+                 {JDBCUtils.SYBASE_ASE, "TOP 0"}
+           });
+    }
+    
+    @Mock
+    Statement statement;
+    
+    @Mock
+    ResultSet resultSet;
+    
+    @Spy
+    SQLDataSetProvider dataSetProvider = SQLDataSetProvider.get();
+    
+    ResultSetHandler resultSetHandler;
```

Comment:
This can be just a local variable. But up to you.

Reply:
Done